packages/superset-ui-legacy-plugin-chart-horizon/src/HorizonChart.jsx

```diff
@@ -78,6 +78,9 @@ class HorizonChart extends React.PureComponent {
       yDomain = d3Extent(allValues, d => d.y);
     }
 
+    if (data.length <= 1)
+      throw new Error('Please select a "Group by" option to enable multiple rows.');
+
     return (
       <div className={`superset-legacy-chart-horizon ${className}`} style={{ height }}>
         {data.map(row => (
```

Comment:
Maybe it's better to add `const groupKeys = Array.isArray(row.key) ? row.key : [row.key]` here.